landoscript/src/landoscript/util/diffs.py

```
@@ -14,7 +14,7 @@ def diff_contents(orig, modified, file):
         # orig does not exist yet; ie: it will be added
         orig_contents = ""
         fromfile = "/dev/null"
-        add_remove_metadata = "new file mode 100644\n"
+        add_remove_metadata = f"new file mode {mode or 100644}\n"
```

Rather than having a default here, this function should raise an error if `mode` is not set when a file is being newly added or removed.